Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -1451,6 +1477,16 @@ DiscusMultipleScatteringCorrection::scatter(const int nScatters, Kernel::PseudoR
         new_vector(shapeObjectWithScatter->material(), k, specialSingleScatterCalc);
   }
 
+  bool considerCollimator = getProperty("RadialCollimator");
+  if (considerCollimator) {
+    const auto &samplePos = detectorInfo.samplePosition();
+    auto hexahedron = createCollimatorHexahedronShape(samplePos, detectorInfo, histogramIndex);
+    // checking is the final scatter point inside the collimatorCorridor shape?
+    if (hexahedron && !hexahedron->isValid(track.startPoint()))
+      return {true, std::vector<double>(wValues.size(), 0.)};
```

Comment:
If `createCollimatorHexahedronShape` comes across an invalid detector shape it returns a `nullptr` do we not also set the weights to zero in this instance? I know from your validation you don't expect this...

Reply:
I also had that doubt and therefore I let the code run as it would do originally without giving 0 weight. 
Yeah it's more sensible to zero them out when `RadialCollimator=True` and the detector shape is not as we expect when the validation at `validateInputs` is removed.